blob/service.go

```diff
@@ -386,6 +522,14 @@ func (s *Service) retrieve(
 		return nil, nil, fmt.Errorf("couldn't find the row index of the namespace end")
 	}
 
+	eds, err := s.shareGetter.GetEDS(ctx, header)
+	if err != nil {
+		return nil, nil, err
```

Comment:
Should we record error in span here?

Reply:
[5534b54](https://github.com/celestiaorg/celestia-node/pull/3633/commits/5534b54173a1ceb124bc9770678d8c967ae96c06)